src/lib.rs

```diff
@@ -333,15 +333,18 @@ pub(crate) unsafe fn rav1d_open(c_out: &mut *mut Rav1dContext, s: &Rav1dSettings
         0 as c_int,
         ::core::mem::size_of::<Rav1dTaskContext>().wrapping_mul((*c).n_tc as usize),
     );
-    if (*c).n_tc > 1 as c_uint {
-        (*c).task_thread.delayed_fg = Mutex::new(mem::zeroed());
-        (*c).task_thread.cond = Condvar::new();
-        (*c).task_thread.cur = AtomicU32::new((*c).n_fc);
-        (*c).task_thread.delayed_fg_cond = Condvar::new();
-        (*c).task_thread.reset_task_cur = AtomicU32::new(u32::MAX);
-        (*c).task_thread.cond_signaled = AtomicI32::new(0);
-        (*c).task_thread.inited = 1 as c_int;
-    }
+    let ttd = TaskThreadData {
+        cond: Condvar::new(),
+        first: AtomicU32::new(0),
+        cur: AtomicU32::new((*c).n_fc),
+        reset_task_cur: AtomicU32::new(u32::MAX),
+        cond_signaled: AtomicI32::new(0),
+        delayed_fg_progress: [AtomicI32::new(0), AtomicI32::new(0)],
+        delayed_fg_cond: Condvar::new(),
+        delayed_fg: Mutex::new(mem::zeroed()),
+        inited: 1,
+    };
+    (&mut (*c).task_thread as *mut Arc<TaskThreadData>).write(Arc::new(ttd));
```

Comment:
Why are these writes to `Arc` fields being done via this awkward pointer write operation (if I'm understanding this correctly)? Is this because the fields are uninitialized and we can't have the destructor run when we overwrite the field?

Reply:
I believe so; that's what I assumed.

Reply:
Correct, I don't want the destructor to run here. That will be removed when `Rav1dContext` is not unsafely zero initialized.